articles/rules/redirect.md

```diff
@@ -29,27 +29,32 @@ function (user, context, callback) {
 }
 ```
 
-Once all rules have finished executing, the user will be redirected to the specified URL.
+Once all rules have finished executing, the user will be redirected to the specified URL. 
+
+Auth0 will also pass a state value in that URL, for example `https://example.com/foo?state=abc123`.
+
+<div class="alert alert-info"><strong>What is state?</strong> State is an opaque value the clients adds to the initial request that Auth0 includes when redirecting back to the client. We highly recommend using this param, since it can be used by the client to prevent <a href="/security/common-threats#cross-site-request-forgery-xsrf-or-csrf-">CSRF attacks</a>.</div>
+
+You need to extract this value since you will need it in order to resume the authentication transaction after the redirect (see `THE_ORIGINAL_STATE` at the next paragraph).
```

Comment:
I think it might be better to take this line out entirely. It's misleading because it follows this sentence:

> State is an opaque value the clients adds to the initial request that Auth0 includes

See below for more.